pallets/utxo/src/lib.rs

```diff
@@ -856,38 +853,32 @@ pub mod pallet {
                 inputs.push(TransactionInput::new_empty(*utxo));
             }
 
-            let pubkey_raw: [u8; 32] = match signer.encode().try_into() {
-                Ok(v) => v,
-                Err(e) => {
-                    log::error!("Failed to get caller's public key: {:?}", e);
-                    return Err("Failed to get caller's public key".into());
-                }
-            };
+            let pubkey_raw: [u8; 32] = signer
+                .encode()
+                .try_into()
+                .map_err(|_| DispatchError::Other("Failed to get caller's public key"))?;
 
-            let pubkey: Public = Public::from_raw(pubkey_raw);
-            let wit_prog = match bech32::wit_prog::WitnessProgram::from_address(
-                address[..2].to_vec(),
-                address,
-            ) {
-                Ok(v) => v,
-                Err(e) => {
-                    log::error!("Failed to decode Bech32 address: {:?}", e);
-                    return Err("Invalid Bech32 address".into());
+            let tx_out = match dest {
+                Destination::Pubkey(pubkey) => {
+                    TransactionOutput::new_pubkey(value, H256::from(pubkey))
                 }
+                Destination::ScriptHash(hash) => TransactionOutput::new_script_hash(value, hash),
+                Destination::CreatePP(code, data) => {
+                    TransactionOutput::new_create_pp(value, code, data)
+                }
+                Destination::CallPP(acc, data) => TransactionOutput::new_call_pp(value, acc, data),
```

Comment:
I think this could be further simplified, just create the TransactionOutput directly (or create a new constructor function to do that):
```rust
TransactionOutput {
  destination: dest, value, header: /* whatever default header is? */
}
```

For a future PR, we should also think about extending these constructors to take output headers directly to make it harder to accidentally mix various token types.

Reply:
I don't understand why I wrote it like that, fixed